### PR TITLE
Adds lidar pattern for raycaster sensor

### PR DESCRIPTION
**Pull request by pascal-roth**
# Description

Add a lidar pattern for the `RayCaster` sensor. The pattern is determined by number of channels, horizontal and vertical field of view and corresponding horizontal resolution. An example config for a Velodyne VLP 16 is provided.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshots

Screenshot from 2024-07-02 10-13-45

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py [hunk 1]
```diff
@@ -122,3 +122,22 @@ class BpearlPatternCfg(PatternBaseCfg):
...
+class LidarPatternCfg(PatternBaseCfg):
+    """Configuration for the LiDAR pattern for ray-casting."""
+
+    func: Callable = patterns.lidar_pattern
+
+    channels: int = MISSING
+    """Number of Channels (Beams). Determines the vertical resolution of the LiDAR sensor."""
+
+    vertical_fov_range: tuple[float, float] = MISSING
+    """Vertical field of view range in degrees."""
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py:137]: NIT: Can we stick to radians for consistency with SI units everywhere? :)

Comment (pascal-roth) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/ray_caster/patterns/patterns_cfg.py:137]: Bpearl pattern cfg is also in degrees, wanted to stay consistent with that one. Also the datasheets was in degrees.

**Review by Dhoeller19 — APPROVED** [review]

**Review by Mayankm96 — APPROVED** [review]
Looks good. Needs update in the docs here: https://github.com/isaac-sim/IsaacLab/edit/main/docs/source/api/lab/omni.isaac.lab.sensors.patterns.rst

File: source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets/velodyne.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Source: https://velodynelidar.com/wp-content/uploads/2019/12/63-9229_Rev-K_Puck-_Datasheet_Web.pdf
+VELODYNE_VLP_16_CFG = RayCasterCfg(
+    attach_yaw_only=False,
+    pattern_cfg=patterns.LidarPatternCfg(
+        channels=16, vertical_fov_range=(-15.0, 15.0), horizontal_fov_range=(-180.0, 180.0), horizontal_res=0.2
+    ),
+    debug_vis=True,
+    max_distance=100,
+)
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets/velodyne.py]:
```suggestion
VELODYNE_VLP_16_CFG = RayCasterCfg(
    attach_yaw_only=False,
    pattern_cfg=patterns.LidarPatternCfg(
        channels=16, vertical_fov_range=(-15.0, 15.0), horizontal_fov_range=(-180.0, 180.0), horizontal_res=0.2
    ),
    debug_vis=True,
    max_distance=100,
)
"""Configuration for Velodyne Puck LiDAR (VLP-16) as a :class:`RayCasterCfg`.

Reference: https://velodynelidar.com/wp-content/uploads/2019/12/63-9229_Rev-K_Puck-_Datasheet_Web.pdf
"""
```

File: source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets/anymal.py [hunk 3]
```diff
@@ -160,3 +163,11 @@
...
     This may impact the sim-to-real transfer performance.
 """
+
+
+##
+# Configuration - Sensors.
+##
+
+ANYMAL_LIDAR_CFG = VELODYNE_VLP_16_CFG.replace(offset=RayCasterCfg.OffsetCfg(pos=(0.0, 0.0, 0.6)))
+"""Configuration for the Velodyne-16 sensor mounted on the ANYmal robot."""
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets/anymal.py]:
```suggestion
"""Configuration for the Velodyne VLP-16 sensor mounted on the ANYmal robot's base."""
```

File: source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets/velodyne.py (hunk 2 above)
Comment (Mayankm96) [reply source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets/velodyne.py]: If we soon plan to have RTX implementation of the sensor, would make sense to make it explicit as raycaster similar to what you did for cameras?

Comment (pascal-roth) [reply source/extensions/omni.isaac.lab_assets/omni/isaac/lab_assets/velodyne.py]: yes, I think that would be smart, will change it now

**Comment by Mayankm96** [comment]
Thanks a lot for the contribution!

**Pull request merged**